keras/metrics.py

```diff
@@ -71,6 +71,25 @@ def cosine_proximity(y_true, y_pred):
     return -K.mean(y_true * y_pred)
 
 
+def matthews(y_true, y_pred):
+    ''' Matthews correlation coefficient
+    '''
+    y_pred_pos = K.round(y_pred)
+    y_pred_neg = 1 - y_pred_pos
```

Comment:
Do you have any ideas on how to enforce this? IIRC y_true and y_pred are tensors, so there's no trivial way of turning these into values, and doing so would hurt performance when this is a training metric.

I could probably come up with some tensor operations to make this get done on the GPU, but it's not fully clear how I would ensure they get validated; in tensorflow it might be possible to hack some hideous assert/group op thing together, but it seems overkill for a simple metric; I have no idea about Theano.